### PR TITLE
Fixing issues leading to build/deployment failures

### LINKED ISSUE #6: Fix the Github Action build failures
Builds triggered as a part of the PRs fail. It seems to be some misconfigured AWS region. Need to fix this to unblock PRs from merging

### PR DESCRIPTION
**Pull request by sanathkr**
Closes #6

- I also ran `npm audit fix` to fix most of the high severity security vulnerabilities. Hence the change to package-lock.json
- Removed Yarn from `package/ui`
- Removed empty files from `package/ui` so `next build` would complete
successfully
- Created a `personal` profile so you can run `npm run deploy:personal`
to deploy to your personal AWS Account instead of the shared account
- Wrote a development guide

**Known Issues**
Looks like Amplify Console does not support deploying Next.js apps that
are have dynamic routes. We have to identify a different deployment
provider. Right now the backend deploys successfully but the frontend
does not.

### CODE REVIEW

**Review by brettstack — APPROVED** [review]

File: serverless.yaml [hunk 1]
```diff
@@ -43,6 +43,15 @@ plugins:
 
 custom:
   stages:
+    personal:
```
Comment (brettstack) [inline serverless.yaml:46]: Can we just use the development stage instead? Should be able to change profile to `profile: ${{env:AWS_PROFILE}}`

**Comment by brettstack** [comment]
Could you elaborate on your dynamic routes comment? If you're talking about it not working when you navigate to /example/route I have a fix for that using Amplifys custom rules.

File: serverless.yaml (hunk 1 above)
Comment (sanathkr) [reply serverless.yaml:46]: oh good idea, i will change that

**Comment by sanathkr** [comment]
@brettstack  When I deployed using `npm run deploy:dev` and navigated to the URL, I got an Access Denied. I checked Amplify Console to ensure the deployment was successful. Turns out there was not index.html generated by Next. 

Digging further, I found [this documentation page](https://docs.amplify.aws/guides/hosting/nextjs/q/platform/js) with the following note

> Note: Next also supports pre-rendering for dynamic server-rendered routes. At this time, Amplify does not support the hosting of dynamic server-rendered routes with Next.

The trick is to use `next build && next export` to generate index.html. My understand is this will prevent us from using dynamic routes. 

Am I correct? Do we actually need these?

**Comment by sanathkr** [comment]
Merged in #11

**Pull request closed without merging**